src/app/pages/details/common/get-this-title-files/give-before-pdf/use-give-dialog.tsx

```diff
@@ -73,12 +72,21 @@ export function useOpenGiveDialog() {
     return {GiveDialog, openGiveDialog};
 }
 
-function lookupVariant(warning: string, variant?: VariantValue, id?: string) {
+// This was a little bit clever, so typing became a problem
+function lookupVariant(warning: string, variantParams: {
+    id?: string;
+    variant?: string;
+}): [
+    Variant: (p: any) => React.JSX.Element, // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
I wonder if a union type that uses ComponentProps would work here but it may not be worth the hassle

Reply:
I fiddled with it, but couldn't make it happy.